core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java

```diff
@@ -638,6 +648,21 @@ public MutationWriter(int numSendThreads) {
       queued = new HashSet<String>();
       sendThreadPool = new SimpleThreadPool(numSendThreads, this.getClass().getName());
       locators = new HashMap<String,TabletLocator>();
+      binTimer.schedule(new TimerTask() {
+        @Override
+        public void run() {
+          MutationSet m = queue.poll();
+          while (null != m) {
+            try {
+              addMutations(m);
+            } catch (Exception e) {
+              updateUnknownErrors("Error processing mutation set", e);
+            }
+            m = queue.poll();
+          }
+        }
+      }, 0, 500);
```

Comment:
Ok I think mutations should be binned and sent immediately.   There are currently two cases when mutations are submitted to be binned and sent: when the users calls flush and when memory 1/2 full.  Don't see a good reason to wait in either case.